perfkitbenchmarker/linux_benchmarks/iperf_benchmark.py

```diff
@@ -98,63 +126,191 @@ def _RunIperf(sending_vm, receiving_vm, receiving_ip_address, thread_count,
+    timeout_buffer = FLAGS.iperf_timeout or 30 + thread_count
+    stdout, _ = sending_vm.RemoteCommand(iperf_cmd, should_log=True,
+                                         timeout=FLAGS.iperf_runtime_in_seconds + timeout_buffer)
+
+    multi_thread = re.search((r'\[SUM\]\s+\d+\.\d+-\d+\.\d+\s\w+\s+(?P<transfer>\d+)\s\w+\s+(?P<throughput>\d+)'
+                              r'\s\w+\/\w+\s+(?P<write>\d+)\/(?P<err>\d+)\s+(?P<retry>\d+)\s*'),
+                             stdout)
+
+    window_size_match = re.search(r'TCP window size: (?P<size>\d+\.?\d+) (?P<units>\S+)', stdout) 
+    window_size = float(window_size_match.group('size'))
+    window_size_unit = window_size_match.group('units')
+
+    buffer_size = float(re.search(r'Write buffer size: (?P<buffer_size>\d+\.\d+) \S+', stdout).group('buffer_size'))
+
+    # Iperf output is formatted differently when running with multiple threads vs a single thread
+    if multi_thread:
+      # Write, error, retry
+      write = int(multi_thread.group('write'))
+      err = int(multi_thread.group('err'))
+      retry = int(multi_thread.group('retry'))
+
+    # if single thread
+    else:
+      # Write, error, retry
+      match = re.search(r'\d+ Mbits\/sec\s+(?P<write>\d+)\/(?P<err>\d+)\s+(?P<retry>\d+)', stdout)
+      write = int(match.group('write'))
+      err = int(match.group('err'))
+      retry = int(match.group('retry'))
+
+    match = re.findall(r'\d+ Mbits\/sec\s+ \d+\/\d+\s+\d+\s+(-*\d+)(\w+)\/(\d+)\s+(\w+)\s+(\d+\.\d+)', stdout)
+
+    cwnd = sum(float(i[0]) for i in match)/len(match)
+    rtt = round(sum(float(i[2]) for i in match)/len(match), 2)
+    netpwr = round(sum(float(i[4]) for i in match)/len(match),2)
+
+    cwnd_unit = match[0][1]
+    rtt_unit = match[0][3]
+
+    thread_values = re.findall(r'\[SUM].*\s+(\d+\.?\d*).Mbits/sec', stdout)
+    if not thread_values:
+      # If there is no sum you have try and figure out an estimate
+      # which happens when threads start at different times.  The code
+      # below will tend to overestimate a bit.
+      thread_values = re.findall(r'\[.*\d+\].*\s+(\d+\.?\d*).Mbits/sec', stdout)
+
+      if len(thread_values) != thread_count:
+        raise ValueError(f'Only {len(thread_values)} out of {thread_count}'
+                         ' iperf threads reported a throughput value.')
+
+    total_throughput = 0.0
+    for value in thread_values:
+      total_throughput += float(value)
+
+    metadata = {
+        # The meta data defining the environment
+        'receiving_machine_type': receiving_vm.machine_type,
+        'receiving_zone': receiving_vm.zone,
+        'sending_machine_type': sending_vm.machine_type,
+        'sending_thread_count': thread_count,
+        'sending_zone': sending_vm.zone,
+        'runtime_in_seconds': FLAGS.iperf_runtime_in_seconds,
+        'ip_type': ip_type,
+        'buffer_size': buffer_size,
+        'tcp_window_size': window_size,
+        'write': write,
```

Comment:
Can you make "write" "err" "retry" "cwnd" "rtt" "rtt_unit" and "netpwr" labels a bit more descriptive?  I'm not sure what they represent.

For example based on the test of "0.00-60.00 sec  15.0 MBytes  2.10 Mbits/sec  10700/0      178 pps" it could be something like this:

write_packet_count=10700
err_packet_count= 0
retry_packet_count = ?

I realize that it looks pretty verbose but putting "_packet_count" in the label makes it a lot easier to tell what is going on when we look at this in a year.